cpp/src/table/legacy/table.cpp

```diff
@@ -27,12 +27,12 @@
 namespace cudf {
 
 table::table(std::vector<gdf_column*> const& cols) : _columns{cols} {
-  CUDF_EXPECTS(_columns.size() != 0, "Cannot construct table with zero columns");
-  std::for_each(_columns.begin(), _columns.end(),
-                [this](gdf_column* col) {
-                  CUDF_EXPECTS(nullptr != col, "Null input column");
-                  CUDF_EXPECTS(_columns.front()->size == col->size, "Column size mismatch");
-                });
+  if (not cols.empty())
```

Comment:
The way the `for_each` is written, there's no need for this check. It will automatically handle empty vectors.